cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -739,8 +739,29 @@ public void deleteGenericResourceById(String databaseServerId) {
         handleAuthException(() -> azure.genericResources().deleteById(databaseServerId));
     }
 
-    public PrivateZone getPrivateDnsZoneByResourceGroup(String resourceGroupName, String dnsZoneName) {
-        return privatednsManager.privateZones().getByResourceGroup(resourceGroupName, dnsZoneName);
+    public PagedList<PrivateZone> getPrivateDnsZoneList() {
+        return privatednsManager.privateZones().list();
+    }
+
+    public String validateNetworkLinkExistenceForDnsZones(String networkLinkId, List<AzurePrivateDnsZoneServiceEnum> services) {
+        PagedList<PrivateZone> privateDnsZoneList = getPrivateDnsZoneList();
+        for (AzurePrivateDnsZoneServiceEnum service : services) {
+            String dnsZoneName = service.getDnsZoneName();
+            Optional<PrivateZone> privateZoneWithNetworkLink = privateDnsZoneList.stream()
+                    .filter(privateZone -> privateZone.name().equals(dnsZoneName))
+                    .filter(privateZone -> privateZone.provisioningState().equals(SUCCEEDED))
+                    .filter(privateZone -> Objects.nonNull(getNetworkLinkByPrivateDnsZone(privateZone.resourceGroupName(), dnsZoneName, networkLinkId)))
+                    .findFirst();
+            if (privateZoneWithNetworkLink.isPresent()) {
+                PrivateZone privateZone = privateZoneWithNetworkLink.get();
+                String validationMessage = String.format("Network link for the network %s already exists for Private DNS Zone %s in resource group %s. "
+                            + "Please ensure that there is no existing network link and try again!",
+                    networkLinkId, dnsZoneName, privateZone.resourceGroupName());
+                LOGGER.warn(validationMessage);
+                return validationMessage;
+            }
+        }
+        return null;
```

Comment:
instead of returning `null` please consider a different way, like using `Optional`